src/java/com/reucon/openfire/plugin/archive/xep0313/IQQueryHandler.java

```diff
@@ -265,39 +265,86 @@ private IQ buildForbiddenResponse(IQ packet) {
             }
             if(dataForm.getField("start") != null) {
                 startField = dataForm.getField("start").getFirstValue();
+                Log.debug("MAM: startField="+startField!=null?startField:"NULL");
             }
             if(dataForm.getField("end") != null) {
                 endField = dataForm.getField("end").getFirstValue();
+                Log.debug("MAM: endField="+endField!=null?endField:"NULL");
             }
         }
 
-        Date startDate = null;
-        Date endDate = null;
-        try {
-            if(startField != null) {
-                startDate = xmppDateTimeFormat.parseString(startField);
-            }
-            if(endField != null) {
-                endDate = xmppDateTimeFormat.parseString(endField);
-            }
-        } catch (ParseException e) {
-            Log.error("Error parsing query date filters.", e);
+       try
+        {
+	        long newDate = System.currentTimeMillis()-(conversationManager.getMaxRetrievable() * JiveConstants.DAY);
+	       
+	        Date startDate = null;
+	        Date endDate = null;
+	        try {
+	            if(startField != null) {
+	            	if (conversationManager.getMaxRetrievable()<=0)
+	            	{
+	            		startDate = xmppDateTimeFormat.parseString(startField);
+	            	}
+	            	else
+	            	{
+	            		startDate = new Date(newDate);
+	            	}
+	               
+	            	Log.debug("MAM: startDate="+new SimpleDateFormat("yyyy-MM-dd hh:mm:ss").format(startDate));
+	            }
+	            else
+	            {
+	            	if (conversationManager.getMaxRetrievable()>0)
+	            	{
+	            		startDate = new Date(newDate);
+	            		Log.debug("MAM: startDate="+new SimpleDateFormat("yyyy-MM-dd hh:mm:ss").format(startDate));
+	            	}
+	            	else
+	            	{
+	            		Log.debug("MAM maxRetrievable not set!");
+	            	}
+	            }
+	            if(endField != null) {
+	                endDate = xmppDateTimeFormat.parseString(endField);
+	                Log.debug("MAM: endDate="+endDate!=null?endDate.toString():"NULL");
+	            }
+	            else
+	            {
+	            	endDate=new Date();
+	            }
+	        } catch (ParseException e) {
+	            Log.error("Error parsing query date filters.", e);
+	        }
+	       
+	        Collection <ArchivedMessage> result = getPersistenceManager(queryRequest.getArchive()).findMessages(
+	                startDate,
+	                endDate,
+	                queryRequest.getArchive().toBareJID(),
+	                withField,
+	                queryRequest.getResultSet(),
+                	this.usesUniqueAndStableIDs());
+	        
+	        Log.debug("MAM: found: "+String.valueOf(result.size())+" items");
+	        
+	        return result;
+        }
+        catch (Exception e)
+        {
+        	Log.error("MAM: 0 items found (error): ",e);        	
```

Comment:
It is good to log the exception, but the message that is used is very confusing. By writing "0 items found" there's a suggestion that a search has actually been executed, while that's (probably) not the case. I suggest that you replace this with: "An exception has occurred while retrieving messages."